geonode/geoserver/manager.py

```diff
@@ -0,0 +1,137 @@
+                    geoserver_cascading_delete.apply_async((_real_instance.alternate,))
+
+    def create(self, uuid: str, /, type: object = None, defaults: dict = {}) -> ResourceBase:
+        if type:
+            _resource = type.objects.get(uuid=uuid)
+            if type == Layer:
+                geoserver_post_save_local(_resource)
+            return _resource
+        return None
+
+    def update(self, uuid: str, /, instance: ResourceBase = None, xml_file: str = None, metadata_uploaded: bool = False,
+               vals: dict = {}, regions: dict = {}, keywords: dict = {}, custom: dict = {}, notify: bool = True) -> ResourceBase:
+        if instance:
+            if type(instance.get_real_instance()) == Layer:
+                geoserver_post_save_local(instance.get_real_instance())
+        return instance
+
+    def set_permissions(self, uuid: str, /, instance: ResourceBase = None, permissions: dict = {}, created: bool = False) -> bool:
+        # TODO: move GeoFence set perms logic here
+        return True
+
+    def set_thumbnail(self, uuid: str, /, instance: ResourceBase = None, overwrite: bool = True, check_bbox: bool = True) -> bool:
+        if instance:
+            # TODO: missing thumb for documents
+            geoserver_create_thumbnail.apply_async(((instance.id, overwrite, check_bbox, )))
+            return True
+        return False
+
+    def exec(self, method: str, uuid: str, /, instance: ResourceBase = None, **kwargs) -> ResourceBase:
+        raise NotImplementedError
+
+    def set_style(self, method: str, uuid: str, /, instance: ResourceBase = None, **kwargs) -> ResourceBase:
+        try:
+            instance = instance or Layer.objects.get(uuid=uuid)
+        except Exception as e:
+            logger.exception(e)
+            return None
+
+        if instance and type(instance.get_real_instance()) == Layer:
+            try:
+                logger.info(f'Creating style for Layer {instance.get_real_instance()} / {kwargs}')
+                _sld_file = kwargs['sld_file'] if 'sld_file' in kwargs else None
+                _tempdir = kwargs['tempdir'] if 'tempdir' in kwargs else tempfile.gettempdir()
+                if _sld_file and 'sld_uploaded' in kwargs and kwargs['sld_uploaded'] is True:
+                    geoserver_set_style(instance.get_real_instance().id, _sld_file)
+                else:
+                    geoserver_create_style(instance.get_real_instance().id, instance.get_real_instance().name, _sld_file, _tempdir)
+            except Exception as e:
+                logger.exception(e)
+                return None
+        return instance
+
+    def set_time_info(self, method: str, uuid: str, /, instance: ResourceBase = None, **kwargs) -> ResourceBase:
+        try:
+            instance = instance or Layer.objects.get(uuid=uuid)
+        except Exception as e:
+            logger.exception(e)
+            return None
+
+        if instance and type(instance.get_real_instance()) == Layer:
```

Comment:
Minor: we can also use `isinstance(instance.get_real_instance(), Layer)` if we want to check also for inheritance.